src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -0,0 +1,380 @@
+
+import {bytesToDisplayDataAmount, DisplayDataAmount, displayDataAmountToBytes, formatBytesParts} from '../data_formatting';
+
+import {COMMON_STYLES} from './cloud-install-styles';
+
+/**
+ * A floating window representing settings specific to individual access keys. Its state is
+ * dynamically set when opened using the open() method instead of with any in-HTML attributes.
+ *
+ * This element relies on conceptual separation of the data limit as input by the user, the data
+ * limit of the UI key, and the actual data limit as saved on the server.  App controls the UI data
+ * limit and the request to the server, and the display key in the element is never itself changed.
+ */
+@customElement('outline-per-key-data-limit-dialog')
+export class OutlinePerKeyDataLimitDialog extends LitElement {
+  static get styles() {
+    return [
+      COMMON_STYLES,
+      css`
+        #container {
+          width: 100%;
+          display: flex;
+          flex-flow: column nowrap;
+        }
+
+        #dataLimitIcon {
+          /* Split the padding evenly between the icon and the section to be bidirectional. */
+          padding: 0 12px;
+        }
+
+        #headerSection {
+          display: flex;
+          flex-direction: row;
+          padding: 0 12px;
+        }
+
+        #headerSection h3 {
+          font-size: 18px;
+          font-weight: 500;
+          color: rgba(0, 0, 0, 0.87);
+          line-height: 24px;
+        }
+
+        #menuSection {
+          flex: 1;
+          padding: 0 78px;
+          margin-top: 10px;
+        }
+
+        #buttonsSection {
+          margin-top: 10px;
+          display: flex;
+          flex-direction: row-reverse;
+        }
+
+        paper-button {
+          display: flex;
+          height: 36px;
+          text-align: center;
+        }
+
+        #save {
+          background-color: var(--primary-green);
+          color: #fff;
+        }
+
+        #save[disabled] {
+          color: var(--dark-gray);
+          background-color: rgba(0, 0, 0, 0.13);
+        }
+
+        #menu {
+          display: flex;
+          flex-flow: row nowrap;
+        }
+
+        #unitsDropdown {
+          width: 50px;
+          padding: 0 10px;
+        }
+
+        paper-checkbox {
+          /* We want the ink to be the color we're going to, not coming from */
+          --paper-checkbox-checked-color: var(--primary-green);
+          --paper-checkbox-checked-ink-color: var(--dark-gray);
+          --paper-checkbox-unchecked-color: var(--dark-gray);
+          --paper-checkbox-unchecked-ink-color: var(--primary-green);
+        }
+
+        paper-listbox paper-item:hover {
+          cursor: pointer;
+          background-color: var(--background-contrast-color);
+          color: #fff;
+        }
+      `,
+    ];
+  }
+
+  /**
+   * @member _keyName The displayed name of the UI access key representing the key we're working on.
+   */
+  @internalProperty() _keyName = '';
```

Comment:
I don't understand how we'd do that in a useful way -- `keyName` isn't known at the time the dialog is constructed and added to the dom, so we'd still need to set it in `open()` anyways.  Alternatively we make the dialog immutable and destroy and re-create new ones instead of opening and closing a singleton, but even in that case we'd have to pass `keyName` to the constructor.

Reply:
See below